dataplane/linux/vxlan_mgr_test.go

```diff
@@ -0,0 +1,203 @@
+	l := []netlink.Addr{{
+		IPNet: &net.IPNet{
+			IP: net.IPv4(172, 0, 0, 2),
+		},
+	},
+	}
+	return l, nil
+}
+
+func (mockVXLAND *mockVXLANDataplan) AddrAdd(link netlink.Link, addr *netlink.Addr) error {
+	return nil
+}
+
+func (mockVXLAND *mockVXLANDataplan) AddrDel(link netlink.Link, addr *netlink.Addr) error {
+	return nil
+}
+
+func (mockVXLAND *mockVXLANDataplan) LinkList() ([]netlink.Link, error) {
+	return mockVXLAND.links, nil
+}
+
+func (mockVXLAND *mockVXLANDataplan) LinkAdd(netlink.Link) error {
+	return nil
+}
+func (mockVXLAND *mockVXLANDataplan) LinkDel(netlink.Link) error {
+	return nil
+}
+
+var _ = Describe("VXLANManager", func() {
+	var manager *vxlanManager
+	var rt *mockRouteTable
+	var prt *mockRouteTable
+
+	BeforeEach(func() {
+		rt = &mockRouteTable{
+			currentRoutes:   map[string][]routetable.Target{},
+			currentL2Routes: map[string][]routetable.L2Target{},
+		}
+		prt = &mockRouteTable{
+			currentRoutes:   map[string][]routetable.Target{},
+			currentL2Routes: map[string][]routetable.L2Target{},
+		}
+
+		manager = newVXLANManagerWithShims(
+			newMockIPSets(),
+			rt,
+			"vxlan.calico",
+			Config{
+				MaxIPSetSize:       5,
+				Hostname:           "node1",
+				ExternalNodesCidrs: []string{"10.0.0.0/24"},
+				RulesConfig: rules.Config{
+					VXLANVNI:  1,
+					VXLANPort: 20,
+				},
+			},
+			&mockVXLANDataplan{
+				links: []netlink.Link{&mockLink{attrs: netlink.LinkAttrs{Name: "eth0"}}},
+			},
+			func(interfacePrefixes []string, ipVersion uint8, vxlan bool, netlinkTimeout time.Duration,
+				deviceRouteSourceAddress net.IP, deviceRouteProtocol int, removeExternalRoutes bool) routeTable {
+				return prt
+			},
+		)
+	})
+
+	It("successfully adds a route to the parent interface", func() {
+		manager.OnUpdate(&proto.VXLANTunnelEndpointUpdate{
+			Node:           "node1",
+			Mac:            "00:0a:74:9d:68:16",
+			Ipv4Addr:       "10.0.0.0",
+			ParentDeviceIp: "172.0.0.2",
+		})
+
+		manager.OnUpdate(&proto.VXLANTunnelEndpointUpdate{
+			Node:           "node2",
+			Mac:            "00:0a:95:9d:68:16",
+			Ipv4Addr:       "10.0.80.0/32",
+			ParentDeviceIp: "172.0.12.1",
+		})
+
+		localVTEP := manager.getLocalVTEP()
+		Expect(localVTEP).NotTo(BeNil())
+
+		manager.parentRouteTable = prt
+
+		err := manager.configureVXLANDevice(50, localVTEP)
+		Expect(err).To(BeNil())
+
+		Expect(manager.myVTEP).NotTo(BeNil())
+		Expect(manager.parentRouteTable).NotTo(BeNil())
+		parent, err := manager.getLocalVTEPParent()
+
+		Expect(parent).NotTo(BeNil())
+		Expect(err).To(BeNil())
+
+		manager.OnUpdate(&proto.RouteUpdate{
+			Type: proto.RouteType_DEFAULT,
+			Node: "node2",
+			Dst:  "172.0.0.1/32",
+			Gw:   "172.8.8.8/32",
+		})
+
+		manager.OnUpdate(&proto.RouteUpdate{
+			Type: proto.RouteType_VXLAN,
+			Node: "node2",
+			Dst:  "172.0.0.2/32",
+			Gw:   "172.8.8.8/32",
+		})
+
+		err = manager.CompleteDeferredWork()
+
+		Expect(err).To(BeNil())
+		Expect(rt.currentRoutes["vxlan.calico"]).NotTo(BeNil())
```

Comment:
Might be worth asserting that these are in fact `nil` before calling the deferred work - then you can be confident it was the deferred work execution that caused this change.